code/modules/projectiles/projectile.dm

```diff
@@ -178,6 +187,32 @@
 				loc = A.loc
 			permutated.Add(A)
 			return 0
+		else if(!custom_impact)
+			var/obj/effect/overlay/beam/impact/impact = new(get_turf(src))

```

Comment:
Wouldn't it be better to do this as an image() added to overlays of what it hit rather than making an object